Scripts/ResetVmPermissions/FixRoleAssignments.ps1

```diff
@@ -0,0 +1,94 @@
+        return
+    }
+
+    # Give me all labs in the subscription
+    $devTestLabs = Find-AzureRmResource -ResourceType 'Microsoft.DevTestLab/labs'
+
+    foreach ($devTestLab in $devTestLabs) {
+        
+        
+        # Get all the VMs in the lab
+        $virtualMachines = Get-AzureRmResource -ResourceId "$($devTestLab.ResourceId)/virtualmachines" -ApiVersion 2016-05-15 
+
+        foreach ($virtualMachine in $virtualMachines) {
+            # Get the owner of the VM
+
+            $ownerId = $virtualMachine.Properties.ownerObjectId
+            $ownerName= $virtualMachine.Properties.ownerUserPrincipalName
+
+            # the owner can be blank for really old VMs, let's copy created by if machine isn't claimable
+            if (($virtualMachine.Properties.allowClaim -eq $false) -and ($ownerId -eq $null)) {
+                $ownerId = $virtualMachine.Properties.createdByUserId
+                $ownerName = $virtualMachine.Properties.createdByUser
+            }
+
+            # if we have the owner Id, let's check the permissions
+            if (($ownerId -ne $null) -and ($ownerId.Trim().Length -gt 0) ) {
+                
+                $parentResource = ("labs/" + $devTestLab.Name)
+
+                # If the VM role assignment doesn't exist, let's add it back
+                $existing = Get-AzureRmRoleAssignment -ObjectId $ownerId `
+                                                      -RoleDefinitionName "Owner" `
+                                                      -ResourceGroupName $virtualMachine.ResourceGroupName `
+                                                      -ResourceName $virtualMachine.Name `
+                                                      -ResourceType $virtualMachine.ResourceType `
+                                                      -ParentResource $parentResource `
+                                                      | Where-Object {$_.Scope -like "*/virtualMachines/*"}
+                if ($existing -eq $null) {
+                    Write-Output "Fixing missing Virtual Machine Role Assignment, SubId: $($virtualMachine.SubscriptionId), VM RG: $($virtualMachine.ResourceGroupName), VM Name: $($virtualMachine.Name), RoleName: Owner, OwnerId : $ownerId, Owner Email: $ownerName"
+
+                    $temp = New-AzureRmRoleAssignment -ObjectId $ownerId `
+                                                      -RoleDefinitionName "Owner" `
+                                                      -ResourceGroupName $virtualMachine.ResourceGroupName `
+                                                      -ResourceName $virtualMachine.Name `
+                                                      -ResourceType $virtualMachine.ResourceType `
+                                                      -ParentResource $parentResource
+                }
+
+                # if the Lab role assignment doesn't exist, and we're supposed to add it, let's do it
+                if ($LabRoleToAdd -ne "") {

```

Comment:
Could you not combine the above "if ($existing -eq $null)" and set `$LabRoleToAdd` to default to `Owner`? That is, remove the blank value from the set and just default to `Owner`. Just wondering if the two parts of the logic, which seem very similar, could not be combined.